Quantum-Control-Applications/Superconducting/Single Fixed Transmon/IQ_blobs.py

```diff
@@ -114,4 +74,44 @@
 Ie = res_handles.get("I_e").fetch_all()["value"]
 Qe = res_handles.get("Q_e").fetch_all()["value"]
 
-two_state_discriminator(Ig, Qg, Ie, Qe, b_print=True, b_plot=True)
+angle, threshold, fidelity, gg, ge, eg, ee = two_state_discriminator(Ig, Qg, Ie, Qe, b_print=True, b_plot=True)
+
+#########################################
+# The two_state_discriminator gives us the rotation angle which makes it such that all of the information will be in
+# the I axis. This is being done by setting the `rotation_angle` parameter in the configuration.
+# See this for more information: https://qm-docs.qualang.io/guides/demod#rotating-the-iq-plane
+# Once we do this, we can perform active reset using:
+#########################################
+#
+# # Active reset:
+# with if_(I > threshold):
+#     play("pi", "qubit")
+#
+#########################################
+#
+# # Active reset (faster):
+# play("pi", "qubit", condition=I > threshold)
+#
+#########################################
+#
+# # Repeat until success active reset
+# with while_(I > threshold):
+#     play("pi", "qubit")
+#     align("qubit", "resonator")
+#     measure("readout", "resonator", None,
+#                 dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I))
+#
+#########################################
+#
+# # Repeat until success active reset, up to 3 iterations
+# count = declare(int)
```

Comment:
When we declare a value like this, it is automatically assigned to 0?

Reply:
yes, but I added it for clarity